### PR TITLE
fix(ce-work): stop asking about branches before starting work

### PR DESCRIPTION
**Pull request by tmchow**
## Summary

`ce-work` no longer stops to ask about branches before it starts implementing. It applies the same rule #1385 gave `ce-debug`: a branch move that is reversible in one command gets done and announced, not asked about. Previously the setup step could ask three questions before writing a line — rename this opaque branch?, continue here or start a new branch?, and a new-branch / worktree / stay-on-default menu.

Step 2 is now stated as the two conditions that must hold, plus the mechanisms that enforce them:

- **Work lands on a feature branch.** On the default branch, detached, or unable to tell → create one from `HEAD` named from the plan and say so (a spare branch is the safe direction). Otherwise continue on the branch you were invoked on. No rename, no menu; a worktree or the default branch only when the user asked this session.
- **Nothing the user did not offer is committed or published.** Step 2 records what was dirty and unpushed at start. Untouched WIP never enters a commit (unit commits are path-limited; the ship handoff passes those files as `exclude:`); pre-existing unpushed commits keep the ship local via `ce-commit`, mirroring `ce-debug`. The one surviving question is a unit that must edit a file the user already had dirty — asked once, at the first commit that would include it. In return-to-caller mode (`lfg`) that case blocks before the edit instead of asking.

Supporting changes in the callees: `ce-commit` and `ce-commit-push-pr` honor `exclude:<paths>` and commit with a trailing path list so pre-staged index entries never ride into a commit; stack construction does the same.

## Validation

`bun run test`: 3117 pass, 0 fail. The change is skill prose plus the callee `exclude:` / path-limited commit contract; no test pinned the old menu.

## Security Disclosure
No security-relevant changes.

## Agent Disclosure
- **Model:** `Claude Code · claude-fable-5`

### CODE REVIEW

File: skills/ce-work/SKILL.md [hunk 1]
```diff
@@ -135,47 +137,14 @@ Determine how to proceed based on what was provided in `<input_document>` (after
...
...cord the pre-work scope** — the files `git status --short` already lists — before editing anything. That list is the user's in-progress work; incremental commits stage only work-owned files, so untouched pre-existing dirt never enters a commit. It rides along on any branch move (`git checkout -b` and `git branch -m` both carry uncommitted changes), which is the intended outcome — do not stash it, do not ask about it, and do not derive the branch name from it: name the branch from the plan or work description regardless of what is dirty. Where a unit later edits a file that was already dirty at start, one commit cannot separate their edits from yours (`ce-commit` and the incremental-commit rule group at file level). Ask once — one question covering every such file, at the first commit that would include one — whether to commit those files with their pre-existing edits included, or leave them uncommitted for the user to handle. That is the single branch/tree question this skill asks, because both answers lose something the agent cannot choose for the user. In Return-to-Caller Mode do not ask: commit them with the unit and name those files in the envelope so the caller can surface it.
```
[the remark this answers is not in the document]
Comment (tmchow) [reply skills/ce-work/SKILL.md]: > `lfg` never reads or reports that field.

Fixed in c6be1ce6 — `lfg` step 2 now reads `pre_existing_edits_committed`, step 6 adds a bullet per file to the run-report comment, and step 8 passes the list into the PR-description context, mirroring how `settled_decision_conflicts` travels.

[the remark this answers is not in the document]
Comment (tmchow) [reply skills/ce-work/SKILL.md]: > The leave-uncommitted path must terminate shipping or carry an enforceable exclusion through every later commit path.

Fixed in c6be1ce6 — "leave them uncommitted" is now an exclusion for the rest of the run: the files are passed as excluded on the `ce-commit-push-pr` / `ce-commit` invocation and named in the final summary as unshipped work-owned changes.

File: skills/ce-work/SKILL.md [hunk 2]
```diff
@@ -135,47 +137,14 @@ Determine how to proceed based on what was provided in `<input_document>` (after
...
-   - Never commit directly to the default branch without explicit permission
+   **If on the default branch:** create a feature branch without asking — `git checkout -b <name>`, named from the plan title or work description (e.g., `feat/user-authentication`, `fix/email-validation`) — and say which branch you moved to. If the user asked for a worktree in this session, invoke the `ce-worktree` skill instead of branching in place. Continue on the default branch only when the user explicitly said so this session.
 
-   **Recommendation**: Use worktree if:
-   - You want to work on multiple features simultaneously
-   - You want to keep the default branch clean while experimenting
-   - You plan to switch between branches frequently
+   **If already on a feature branch:** continue on it. If its name is opaque or auto-generated (for example `worktree-jolly-beaming-raven`) and it has no upstream (`git rev-parse --abbrev-ref @{upstream}` fails), rename it to a name derived from the work with `git branch -m <meaningful-name>` and say so. Leave the name alone when the branch has an upstream — a rename after publication forks the local and remote names, which is not a one-command undo.
```
[the remark this answers is not in the document]
Comment (tmchow) [reply skills/ce-work/SKILL.md]: > resolve local `main`/`master` and nonstandard defaults before taking the automatic feature-branch path.

Fixed in c6be1ce6 — the fallback now takes the first of `main`/`master` that exists on origin *or* locally, and a `current_branch` named `main`/`master` is treated as the default even when the fallback resolves nothing.

[the remark this answers is not in the document]
Comment (tmchow) [reply skills/ce-work/SKILL.md]: > Create and re-read a feature branch before implementation in this state

Fixed in c6be1ce6 — an empty `current_branch` (detached HEAD) now takes the same create-a-feature-branch path as the default branch, and the step re-reads `git branch --show-current` afterwards.

File: skills/ce-work/SKILL.md [hunk 3]
```diff
@@ -125,57 +125,30 @@ Determine how to proceed based on what was provided in `<input_document>` (after
...
...sh it, do not ask about it, and do not derive the branch name from it: name the branch from the plan or work description regardless of what is dirty. Where a unit later edits a file that was already dirty at start, one commit cannot separate their edits from yours (`ce-commit` and the incremental-commit rule group at file level). Ask once — one question covering every such file, at the first commit that would include one — whether to commit those files with their pre-existing edits included, or leave them uncommitted for the user to handle. That is the single branch/tree question this skill asks, because both answers lose something the agent cannot choose for the user. "Leave them uncommitted" is an exclusion that holds for the rest of the run: those files stay out of every later commit, including the shipping handoff — pass them as excluded files on the `ce-commit-push-pr` / `ce-commit` invocation, since either skill would otherwise commit the still-dirty file at file granularity — and name them in the final summary as work-owned changes left unshipped. In Return-to-Caller Mode do not ask: commit them with the unit and name those files in the envelope so the caller can surface it.
```
[the remark this answers is not in the document]
Comment (tmchow) [reply skills/ce-work/SKILL.md]: > the subsequent plain `git commit` in Phase 2 commits every entry already in the index, including untouched pre-existing files.

Fixed in 19b826ec — the incremental-commit workflow now path-limits the commit (`git commit -m … -- <unit files>`) so anything the user had staged before the run stays in the index.

File: skills/ce-work/SKILL.md [hunk 4]
```diff
@@ -125,57 +125,30 @@ Determine how to proceed based on what was provided in `<input_document>` (after
...
 
-   **Recommendation**: Use worktree if:
-   - You want to work on multiple features simultaneously
-   - You want to keep the default branch clean while experimenting
-   - You plan to switch between branches frequently
+   **If already on a feature branch:** continue on it. If its name is opaque or auto-generated (for example `worktree-jolly-beaming-raven`) and it has no upstream (`git rev-parse --abbrev-ref @{upstream}` fails), rename it to a name derived from the work with `git branch -m <meaningful-name>` and say so. Leave the name alone when the branch has an upstream — a rename after publication forks the local and remote names, which is not a one-command undo.
```
[the remark this answers is not in the document]
Comment (tmchow) [reply skills/ce-work/SKILL.md]: > confirm remote/PR publication rather than relying only on `@{upstream}`.

Fixed in 19b826ec — the rename condition is now "unpublished": `git ls-remote --heads origin <branch>` prints nothing. A missing upstream is explicitly called out as not the test.

File: skills/ce-work/SKILL.md [hunk 5]
```diff
@@ -125,57 +125,30 @@ Determine how to proceed based on what was provided in `<input_document>` (after
...
    default_branch=$(git symbolic-ref refs/remotes/origin/HEAD 2>/dev/null | sed 's@^refs/remotes/origin/@@')
 
-   # Fallback if remote HEAD isn't set
+   # Fallback if remote HEAD isn't set: the first of main/master that exists on origin or locally
    if [ -z "$default_branch" ]; then
-     default_branch=$(git rev-parse --verify origin/main >/dev/null 2>&1 && echo "main" || echo "master")
+     for candidate in main master; do
+       if git show-ref --verify --quiet "refs/remotes/origin/$candidate" || git show-ref --verify --quiet "refs/heads/$candidate"; then
+         default_branch=$candidate; break
+       fi
```
[the remark this answers is not in the document]
Comment (tmchow) [reply skills/ce-work/SKILL.md]: > A leftover local `main` can win over remote `master` when `origin/HEAD` is unset

Fixed in 19b826ec — the fallback now checks `origin/main`, `origin/master` first, and only then local `main`/`master` (the local-only repo case).

File: skills/ce-work/SKILL.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (tmchow) [reply skills/ce-work/SKILL.md]: > Neither skill documents or honors an exclusion argument

Fixed in 19b826ec — `ce-commit` and `ce-commit-push-pr` now both document and honor `exclude:<paths>` on the invocation (never stage or commit those files, report them as left out), and `ce-work` passes the leave-uncommitted files in that form.

File: skills/lfg/SKILL.md [hunk 6]
```diff
@@ -70,7 +70,7 @@ When the implementation instruction instead names an ordered fallback list, do n
...
+   When `behavior_change: true`, also require `verification_evidence` that names the relevant units/tasks, existing tests inspected, tests added/changed or used unchanged, red failure or characterization evidence when applicable, verification run, and any deliberate test exception. Do NOT decide the test strategy inside LFG; the evidence is ce-work's contract. Also read `settled_decision_conflicts` from the return: blocker-routed entries arrive as `status: blocked` and stop the pipeline; **record any proceeded-and-flagged entries** — they must reach step 6's durable residual record and step 8's PR-description context, since later review may not rediscover them. Also read `pre_existing_edits_committed`: `ce-work` cannot ask in this mode, so any file listed there carried the user's own uncommitted edits into a unit commit; record a non-empty list the same way, so the run-report comment and PR description say which files it happened to.
```
[the remark this answers is not in the document]
Comment (tmchow) [reply skills/lfg/SKILL.md]: > Reading and recording `pre_existing_edits_committed` sits inside the `behavior_change: true` block.

Fixed in 19b826ec — the read is now stated as unconditional ("Regardless of `behavior_change`, also read …").

File: skills/ce-work/SKILL.md [hunk 7]
```diff
@@ -125,57 +125,28 @@ Determine how to proceed based on what was provided in `<input_document>` (after
...
 
-   **Recommendation**: Use worktree if:
-   - You want to work on multiple features simultaneously
-   - You want to keep the default branch clean while experimenting
-   - You plan to switch between branches frequently
+   **If already on a feature branch:** continue on it. If its name is opaque or auto-generated (for example `worktree-jolly-beaming-raven`) and it is unpublished — no branch of that name on the remote (`git ls-remote --heads origin <current_branch>` prints nothing) — rename it to a name derived from the work with `git branch -m <meaningful-name>` and say so. A missing upstream is not the test: a branch pushed without `-u` has none yet is published. Leave a published branch's name alone — a rename forks the local and remote names, so the shipping tail would push a second branch and miss any open PR, which is not a one-command undo.
```
[the remark this answers is not in the document]
Comment (tmchow) [reply skills/ce-work/SKILL.md]: > a failed `ls-remote` (network, auth, or missing `origin`) also yields empty stdout.

Fixed in e87832a6 — the probe is now `git ls-remote --exit-code --heads origin <branch>` and only exit 2 (no matching ref) permits the rename; exit 0 is published and any other non-zero is unknown, which keeps the name.

[the remark this answers is not in the document]
Comment (tmchow) [reply skills/ce-work/SKILL.md]: > treat nonzero as unknown and leave the name unchanged or stop.

Fixed in e87832a6 — exactly that: `--exit-code`, rename only on exit 2, every other exit status keeps the name.

File: skills/ce-work/SKILL.md [hunk 8]
```diff
@@ -285,10 +256,12 @@ Before implementing the first task, you must read `references/implementation-loo
    # 2. Stage only files related to this logical unit (not `git add .`)
    git add <files related to this logical unit>
 
-   # 3. Commit with conventional message
-   git commit -m "feat(scope): description of this unit"
+   # 3. Commit with conventional message, limited to those same paths
+   git commit -m "feat(scope): description of this unit" -- <files related to this logical unit>
    ```
 
+   The path limit on `git commit` is load-bearing: a bare `git commit` takes the whole index, so anything the user had staged before this run started (Phase 1 Step 2's pre-work scope) would ride into the unit's commit. Naming the paths commits only them and leaves the user's staged entries in the index.
```
[the remark this answers is not in the document]
Comment (tmchow) [reply skills/ce-work/SKILL.md:248]: > `ce-commit` and `ce-commit-push-pr` still run a bare `git commit` after staging named files.

Fixed in e87832a6 — both skills' commit command is now path-limited (`git commit -m … -- file1 file2 file3`) with a note on why, so pre-staged index entries and `exclude:` paths stay out of the handoff commit.

File: skills/ce-commit-push-pr/SKILL.md [hunk 9]
```diff
@@ -86,7 +86,7 @@ If on the default branch, branch creation needs to handle stale local `<base>`,
...
 
-Stage and commit each group. **Avoid `git add -A` and `git add .`** — they sweep in `.env`, build artifacts, and generated files. When a plan Implementation Unit ID is already in hand for this commit (conversation, caller, or the files belong to one unit), append that unit's U-ID in parentheses — `(U3)` means unit 3. Do not hunt for a plan. Omit when the commit spans units, the unit is unclear, or no plan is in hand.
+Stage and commit each group. **Avoid `git add -A` and `git add .`** — they sweep in `.env`, build artifacts, and generated files. **Honor `exclude:<paths>` when the invocation carries it:** a caller names files that must stay uncommitted (typically a user's own in-progress edits it could not separate from its work); never stage or commit them, and say in the report that they were left out. When a plan Implementation Unit ID is already in hand for this commit (conversation, caller, or the files belong to one unit), append that unit's U-ID in parentheses — `(U3)` means unit 3. Do not hunt for a plan. Omit when the commit spans units, the unit is unclear, or no plan is in hand.
```
[the remark this answers is not in the document]
Comment (tmchow) [reply skills/ce-commit-push-pr/SKILL.md:89]: > use a path-limited commit or otherwise preserve excluded index entries.

Fixed in e87832a6 — `ce-commit-push-pr` and `ce-commit` now commit with a trailing path list, so only the named group is committed and other index entries are left alone.

File: skills/ce-work/SKILL.md [hunk 10]
```diff
@@ -125,57 +125,28 @@ Determine how to proceed based on what was provided in `<input_document>` (after
...
 
-   **Option C: Continue on the default branch**
-   - Requires explicit user confirmation
-   - Only proceed after user explicitly says "yes, commit to [default_branch]"
-   - Never commit directly to the default branch without explicit permission
+   **If on the default branch, or detached (`current_branch` is empty):** create a feature branch without asking — `git checkout -b <name>`, named from the plan title or work description (e.g., `feat/user-authentication`, `fix/email-validation`) — and say which branch you moved to; re-read `git branch --show-current` afterwards so the rest of the run uses the new name. A `current_branch` equal to `main` or `master` is the default branch even when the fallback above resolved nothing, as in a local-only repo. If the user asked for a worktree in this session, invoke the `ce-worktree` skill instead of branching in place. Continue on the default branch only when the user explicitly said so this session.
```
[the remark this answers is not in the document]
Comment (tmchow) [reply skills/ce-work/SKILL.md]: > unrelated commits already on local `main`/`master` become ancestors of the implementation commits and are included in the eventual PR.

Fixed differently in 6140383f — the branch still starts from `HEAD` (so the idempotency check keeps seeing those commits), but Step 2 now records whether `HEAD` is ahead of the remote default, and Phase 4's ship step commits locally via `ce-commit` instead of pushing when pre-existing unpushed commits are on the branch (the same "already pushed is not already offered" rule `ce-debug` adopted in #1385). Nothing unoffered reaches a PR without the user's word.

[the remark this answers is not in the document]
Comment (tmchow) [reply skills/ce-work/SKILL.md]: > resolve the remote default through another authoritative source or create a branch conservatively when it remains unknown.

Fixed in 6140383f — the enumerated bash fallback is gone; the step now states the condition: `origin/HEAD`, else the host's reported default (`gh repo view --json defaultBranchRef`), else `main`/`master`, and when still unknown, treat the current branch as the default and create a feature branch (the safe direction).

File: skills/ce-commit-push-pr/SKILL.md [hunk 11]
```diff
@@ -86,15 +86,17 @@ If on the default branch, branch creation needs to handle stale local `<base>`,
...
```
[the remark this answers is not in the document]
Comment (tmchow) [reply skills/ce-commit-push-pr/SKILL.md:89]: > propagate the exclusion into stack construction or stop before constructing the stack.

Fixed in 6140383f — `stack-submit.md` retrospective construction now leaves `exclude:` paths out of the change set entirely and path-limits every layer commit.

File: skills/ce-work/SKILL.md [hunk 12]
```diff
@@ -125,57 +125,15 @@ Determine how to proceed based on what was provided in `<input_document>` (after
...
-   - Never commit directly to the default branch without explicit permission
-
-   **Recommendation**: Use worktree if:
-   - You want to work on multiple features simultaneously
-   - You want to keep the default branch clean while experimenting
-   - You plan to switch between branches frequently
+   **If already on a feature branch:** continue on it. If its name is opaque or auto-generated (for example `worktree-jolly-beaming-raven`) and you can positively confirm it was never published — `git ls-remote --exit-code --heads origin <current_branch>` exits 2 (no matching ref) — rename it to a name derived from the work with `git branch -m <meaningful-name>` and say so. Any other result keeps the name: exit 0 means published, and a non-zero exit other than 2 (unreachable remote, auth failure, no `origin`) means unknown, which is not a license to rename. A missing upstream is not the test either: a branch pushed without `-u` has none yet is published. Leave a published branch's name alone — a rename forks the local and remote names, so the shipping tail would push a second branch and miss any open PR, which is not a one-command undo.
```
[the remark this answers is not in the document]
Comment (tmchow) [reply skills/ce-work/SKILL.md]: > The branch publication probe uses an unquoted placeholder in the command template

Fixed in a5ca5eef — the placeholder is quoted and the prose says why.

File: skills/ce-commit-push-pr/references/stack-submit.md [hunk 13]
```diff
@@ -37,13 +37,13 @@ When one safe topology is clear, proceed without asking: explicit stack intent a
...
+For uncommitted whole-file groups on an existing feature branch, save all tracked and untracked working changes before switching branches, then restore them only on the planned layer whose parent contains their prerequisites; keep the saved work until the constructed top is verified complete. Initialize or adopt the bottom layer at the resolved `<base>` tip or its planned commit tip, commit only its files, then add and commit each next layer in order. Files named by an `exclude:<paths>` carrier on the invocation belong to no layer: leave them out of the change set entirely, and path-limit every layer commit so a pre-staged excluded file cannot ride in. Compose `<bottom-message>` and `<next-message>` with the same subject rule as Step 3: when a plan Implementation Unit ID is already in hand for that layer's commit, append that unit's U-ID in parentheses — `(U3)` means unit 3. Do not hunt for a plan. Omit when the commit spans units, the unit is unclear, or no plan is in hand.
```
[the remark this answers is not in the document]
Comment (tmchow) [reply skills/ce-commit-push-pr/references/stack-submit.md]: > Nothing restores those excluded paths before the save is dropped after verification

Fixed in a5ca5eef — excluded files are now never saved, moved, or restored (they stay in the working tree as found), the completeness check is "change set minus those files", and a branch switch that would clobber one stops with a residual.

File: skills/ce-work/SKILL.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (tmchow) [reply skills/ce-work/SKILL.md]: > A published or meaningful branch name does not establish that its scope matches the new plan.

Fixed in a5ca5eef — continuing on the invoked branch stays the default (invocation from a branch is intent), with one stated exception: a branch whose open PR is plainly a different change from this plan gets a new feature branch from the remote default instead, announced in one line. Shared name or scope is enough to continue, so this doesn't reintroduce the question.

File: skills/ce-work/SKILL.md [hunk 14]
```diff
@@ -125,57 +125,15 @@ Determine how to proceed based on what was provided in `<input_document>` (after
...
...ask about it, and do not derive the branch name from it: name the branch from the plan or work description regardless of what is dirty. Where a unit later edits a file that was already dirty at start, one commit cannot separate their edits from yours (`ce-commit` and the incremental-commit rule group at file level). Ask once — one question covering every such file, at the first commit that would include one — whether to commit those files with their pre-existing edits included, or leave them uncommitted for the user to handle. That is the single branch/tree question this skill asks, because both answers lose something the agent cannot choose for the user. "Leave them uncommitted" is an exclusion that holds for the rest of the run: those files stay out of every later commit, including the shipping handoff — pass them as `exclude:<paths>` on the `ce-commit-push-pr` / `ce-commit` invocation (both honor it; without it either skill would commit the still-dirty file at file granularity) — and name them in the final summary as work-owned changes left unshipped. In Return-to-Caller Mode do not ask: commit them with the unit and name those files in the envelope so the caller can surface it.
```
[the remark this answers is not in the document]
Comment (tmchow) [reply skills/ce-work/SKILL.md]: > a headless run should return a blocker or preserve an exclusion for the caller rather than choosing to commit the user's edits.

Fixed in a5ca5eef — return-to-caller mode now leaves the overlapping files uncommitted and returns `status: blocked` naming them; the user clears it by committing or stashing and rerunning. The `pre_existing_edits_committed` envelope field and its `lfg` plumbing are removed since nothing is committed anymore.

File: skills/ce-commit/SKILL.md [hunk 15]
```diff
@@ -43,15 +43,17 @@ Treat this as a snapshot. Re-read branch and staged set immediately before commi
...
+6. **Stage and commit** — stage **named files only** (never `git add -A` or `git add .`). Honor `exclude:<paths>` when the invocation carries it: those files stay uncommitted no matter what else changed; say in the report that they were left out. Prefer one shell call per commit group:
 
 ```bash
 git add file1 file2 file3 && git commit -m "$(cat <<'EOF'
 type(scope): subject line here
 
 Optional body when the why is not obvious from the subject.
 EOF
-)"
+)" -- file1 file2 file3
```
[the remark this answers is not in the document]
Comment (tmchow) [reply skills/ce-commit/SKILL.md:54]: > The new path-limited commit example appends file placeholders unquoted (`-- file1 file2 file3`).

Not addressing — `file1 file2 file3` are illustrative placeholders in a template, in the same unquoted form the `git add file1 file2 file3` on the same line has always used; the agent composes the real command with real paths and quotes as needed. Quoting only the trailing list would make the example inconsistent without changing what an agent does.

File: skills/ce-work/SKILL.md [hunk 16]
```diff
@@ -125,57 +125,15 @@ Determine how to proceed based on what was provided in `<input_document>` (after
...
..., one commit cannot separate their edits from yours (`ce-commit` and the incremental-commit rule group at file level). Ask once — one question covering every such file, at the first commit that would include one — whether to commit those files with their pre-existing edits included, or leave them uncommitted for the user to handle. That is the single branch/tree question this skill asks, because both answers lose something the agent cannot choose for the user. "Leave them uncommitted" is an exclusion that holds for the rest of the run: those files stay out of every later commit, including the shipping handoff — pass them as `exclude:<paths>` on the `ce-commit-push-pr` / `ce-commit` invocation (both honor it; without it either skill would commit the still-dirty file at file granularity) — and name them in the final summary as work-owned changes left unshipped. In Return-to-Caller Mode do not ask and do not commit them: leave those files uncommitted and return `status: blocked` naming them and why (the caller runs headless and would push whatever you commit; the user's uncommitted edits are not yours to publish). The user clears it by committing or stashing their edits and rerunning.
```
[the remark this answers is not in the document]
Comment (tmchow) [reply skills/ce-work/SKILL.md]: > That decision is still timed like the interactive ask — at the first commit after those files were already edited

Fixed in the latest push — return-to-caller mode now decides before the edit: when a unit is about to touch a file that was dirty at start, it does not edit it and returns `status: blocked` naming the file, so the user's WIP is never mixed and commit/stash-and-rerun is a clean recovery.

File: skills/ce-work/SKILL.md [hunk 17]
```diff
@@ -125,57 +125,15 @@ Determine how to proceed based on what was provided in `<input_document>` (after
...
...re invoked on is where the work belongs. The one exception is a branch whose open PR is plainly a different change from this plan (compare the PR's title and diff to the plan; a shared name or scope is enough to continue): then create the new feature branch from the remote default instead of adding unrelated commits to that PR, and say so. If its name is opaque or auto-generated (for example `worktree-jolly-beaming-raven`) and you can positively confirm it was never published — `git ls-remote --exit-code --heads origin "<current_branch>"` (quote the name — branch names can carry shell metacharacters) exits 2 (no matching ref) — rename it to a name derived from the work with `git branch -m <meaningful-name>` and say so. Any other result keeps the name: exit 0 means published, and a non-zero exit other than 2 (unreachable remote, auth failure, no `origin`) means unknown, which is not a license to rename. A missing upstream is not the test either: a branch pushed without `-u` has none yet is published. Leave a published branch's name alone — a rename forks the local and remote names, so the shipping tail would push a second branch and miss any open PR, which is not a one-command undo.
```
[the remark this answers is not in the document]
Comment (tmchow) [reply skills/ce-work/SKILL.md]: > Git refuses that move when dirty files differ at the remote default, and this path gives no fetch, stash, or residual fallback

Fixed in the latest push — the move now fetches the remote default first, and if git refuses the switch because uncommitted files differ, the step stops and says so (WIP stays put; the user commits/stashes or says to continue on the current branch).

File: skills/ce-work/SKILL.md [hunk 18]
```diff
@@ -125,57 +125,15 @@ Determine how to proceed based on what was provided in `<input_document>` (after
...
...at file level). Ask once — one question covering every such file, at the first commit that would include one — whether to commit those files with their pre-existing edits included, or leave them uncommitted for the user to handle. That is the single branch/tree question this skill asks, because both answers lose something the agent cannot choose for the user. "Leave them uncommitted" is an exclusion that holds for the rest of the run: those files stay out of every later commit, including the shipping handoff — pass them as `exclude:<paths>` on the `ce-commit-push-pr` / `ce-commit` invocation (both honor it; without it either skill would commit the still-dirty file at file granularity) — and name them in the final summary as work-owned changes left unshipped. In Return-to-Caller Mode decide before the edit, not at the commit: when a unit is about to touch a file that was dirty at start, do not edit it — return `status: blocked` naming the file and why (the caller runs headless and would push whatever you commit; the user's uncommitted edits are not yours to publish). Stopping before the edit keeps the user's WIP intact, so committing or stashing it and rerunning is a clean recovery.
```
[the remark this answers is not in the document]
Comment (tmchow) [reply skills/ce-work/SKILL.md]: > Carry the complete pre-work path set into the shipping boundary or otherwise constrain the callee to work-owned files.

Fixed in the latest push — Phase 4 now passes every pre-work file this run did not commit (untouched WIP and leave-uncommitted files alike) as `exclude:<paths>` on the `ce-commit-push-pr` invocation, and Step 2 says so.

[the remark this answers is not in the document]
Comment (tmchow) [reply skills/ce-work/SKILL.md]: > Capture all untracked paths explicitly before implementation.

Fixed in the latest push — the snapshot is now `git status --short --untracked-files=all`.

File: skills/ce-work/SKILL.md [hunk 19]
```diff
@@ -125,57 +125,15 @@ Determine how to proceed based on what was provided in `<input_document>` (after
...
...fault (`git fetch origin <default>` then `git checkout -b <name> origin/<default>`) instead of adding unrelated commits to that PR, and say so. If git refuses the switch because uncommitted files differ from that tip, stop and say so — the user's WIP stays where it is; they commit or stash it, or tell you to continue on the current branch. If its name is opaque or auto-generated (for example `worktree-jolly-beaming-raven`) and you can positively confirm it was never published — `git ls-remote --exit-code --heads origin "<current_branch>"` (quote the name — branch names can carry shell metacharacters) exits 2 (no matching ref) — rename it to a name derived from the work with `git branch -m <meaningful-name>` and say so. Any other result keeps the name: exit 0 means published, and a non-zero exit other than 2 (unreachable remote, auth failure, no `origin`) means unknown, which is not a license to rename. A missing upstream is not the test either: a branch pushed without `-u` has none yet is published. Leave a published branch's name alone — a rename forks the local and remote names, so the shipping tail would push a second branch and miss any open PR, which is not a one-command undo.
```
[the remark this answers is not in the document]
Comment (tmchow) [reply skills/ce-work/SKILL.md]: > Return-to-caller mode has no `status: blocked` path here

Fixed in the latest push — that stop is now explicitly `status: blocked` (naming the branch and files) in Return-to-Caller Mode, never a continuation on the wrong PR branch.

File: skills/ce-work/SKILL.md [hunk 20]
```diff
@@ -125,57 +125,13 @@ Determine how to proceed based on what was provided in `<input_document>` (after
...
-   ```bash
-   git branch -m <meaningful-name>
-   ```
-   Derive the new name from the plan title or work description (e.g., `feat/crowd-sniff`). Present the rename as a recommended option alongside continuing as-is.
-
-   Then ask: "Continue working on `[current_branch]`, or create a new branch?"
-   - If continuing (with or without rename), proceed to step 3
-   - If creating new, follow Option A or B below
+   **Branch.** Determine the default branch (`origin/HEAD`; else what the host reports as the default, e.g. `gh repo view --json defaultBranchRef`; else `main`/`master` when one exists). If you are on it, detached, or cannot tell, create a feature branch from `HEAD` named from the plan or work description (e.g. `feat/user-authentication`) and re-read `git branch --show-current`; the safe direction when unsure is a spare branch, never incremental commits on the real default. Otherwise continue on the branch you were invoked on — that is where the work belongs; do not rename it and do not ask whether to. A worktree is used only when the user asked for one this session (`ce-worktree`); the default branch is committed to only when the user explicitly said so this session.
```
[the remark this answers is not in the document]
Comment (tmchow) [reply skills/ce-work/SKILL.md]: > That removes the unpublished opaque-name rename

Declined — that removal is deliberate. Auto-rename was cosmetic (the PR title carries the meaning) and was the source of most of this PR's review churn (publication probes, exit codes, quoting). The PR description has been updated to match; the summary you reviewed predates that refresh.

[the remark this answers is not in the document]
Comment (tmchow) [reply skills/ce-work/SKILL.md]: > require a verified fresh base when the local default has no commits that need carrying.

Fixed in the latest push — the new branch is based on `origin/<default>` after a fetch when the local default is not ahead of it, and on `HEAD` when it is ahead or there is no remote.

File: skills/ce-work/SKILL.md [hunk 21]
```diff
@@ -125,57 +125,13 @@ Determine how to proceed based on what was provided in `<input_document>` (after
...
-   - If creating new, follow Option A or B below
+   **Branch.** Determine the default branch (`origin/HEAD`; else what the host reports as the default, e.g. `gh repo view --json defaultBranchRef`; else `main`/`master` when one exists). If you are on it, detached, or cannot tell, create a feature branch named from the plan or work description (e.g. `feat/user-authentication`) and re-read `git branch --show-current`. Base it on the fresh remote tip (`git fetch origin <default>`, then `origin/<default>`) when the local default is not ahead of it, so the work builds on current code; base it on `HEAD` when it is ahead (those commits stay visible to the idempotency check, and the shipping gate decides whether they publish) or when there is no remote; the safe direction when unsure is a spare branch, never incremental commits on the real default. Otherwise continue on the branch you were invoked on — that is where the work belongs; do not rename it and do not ask whether to. A worktree is used only when the user asked for one this session (`ce-worktree`); the default branch is committed to only when the user explicitly said so this session.
```
[the remark this answers is not in the document]
Comment (tmchow) [reply skills/ce-work/SKILL.md]: > The keep-`HEAD` vs remote-tip choice keys off whether the local default is ahead, not whether current `HEAD` is.

Fixed in the latest push — the condition now keys off `HEAD` having commits beyond the remote default (which is what Step 2 records), so detached `HEAD` with unpushed commits bases on `HEAD`.
